### PR TITLE
Separate `Options.choice` and `Options.choiceWithValue`

### PR DESCRIPTION
**Pull request by IMax153**

### CODE REVIEW

File: src/Options.ts [hunk 1]
```diff
@@ -128,13 +128,59 @@ export const between: {
...
+ *   ["dog", "cat"]
+ * )
+ *
+ * @since 1.0.0
+ * @category constructors
+ */
+export const choice: <A extends string, C extends NonEmptyReadonlyArray<A>>(
+  name: string,
+  choices: C
+) => Options<{ [K in keyof C]: C[K] }[number]> = internal.choice
```
Comment (tim-smart) [inline src/Options.ts:148]: Can you just do `C[number]` here?

Comment (IMax153) [reply src/Options.ts:148]: Yea - this is leftover from some experimentation with the signature. Will make the change.

File: src/Options.ts [hunk 2]
```diff
@@ -128,13 +128,59 @@ export const between: {
...
+ * export const animal: Options.Options<Animal> = Options.choiceWithValue("animal", [
+ *   ["dog", Dog()],
+ *   ["cat", Cat()],
+ * ])
+ *
  * @since 1.0.0
  * @category constructors
  */
-export const choice: <A, C extends NonEmptyReadonlyArray<readonly [string, A]>>(
+export const choiceWithValue: <A, C extends NonEmptyReadonlyArray<readonly [string, A]>>(
```
Comment (tim-smart) [inline src/Options.ts:180]: I found this doesn't infer A very well.

Instead you can just have the C for choices extending `[string, any][]`, then use `C[number][0]` and `C[number][1]` to extract the types.

Comment (tim-smart) [reply src/Options.ts:180]: LGTM

**Review by tim-smart — APPROVED** [review]

File: src/Options.ts [hunk 3]
```diff
@@ -128,13 +128,59 @@ export const between: {
...
+ *
+ * export const animal: Options.Options<"dog" | "cat"> = Options.choice(
+ *   "animal",
+ *   ["dog", "cat"]
+ * )
+ *
+ * @since 1.0.0
+ * @category constructors
+ */
+export const choice: <A extends string, C extends NonEmptyReadonlyArray<A>>(
```
Comment (tim-smart) [inline src/Options.ts:145]: You can remove A here, and move the string constraint into the array.

Comment (IMax153) [reply src/Options.ts:145]: Tried that first - this doesn't allow the string literals to be inferred:

image

Comment (tim-smart) [reply src/Options.ts:145]: 👍🏻

**Pull request merged**